pymc/tests/test_data.py

```diff
@@ -405,6 +405,18 @@ def test_implicit_coords_dataframe(self):
         assert "columns" in pmodel.coords
         assert pmodel.named_vars_to_dims == {"observations": ("rows", "columns")}
 
+    def test_implicit_coords_xarray(self):
+        xr = pytest.importorskip("xarray")
+        data = xr.DataArray([[1, 2, 3], [4, 5, 6]], dims=("y", "x"))
+        with pm.Model() as pmodel:
+            with pytest.warns(DeprecationWarning):
+                pm.ConstantData("observations", data, dims=("x", "y"), export_index_as_coords=True)
+        assert "x" in pmodel.coords
+        assert "y" in pmodel.coords
+        assert pmodel.named_vars_to_dims == {"observations": ("x", "y")}
+        assert tuple(pmodel.coords["x"]) == (data.coords["x"])
+        assert tuple(pmodel.coords["y"]) == (data.coords["y"])
```

Comment:
I think you'd need this:

```python
        assert tuple(pmodel.coords["x"]) == tuple(data.coords["x"].to_numpy())
        assert tuple(pmodel.coords["y"]) == tuple(data.coords["y"].to_numpy())
```

xarray stores coordinate values as a `DataArray`, so either use the above or `np.testing.assert_array_equal`:

```suggestion
        np.testing.assert_array_equal(pmodel.coords["x"], data.coords["x"])
        np.testing.assert_array_equal(pmodel.coords["y"], data.coords["y"])
```

Reply:
Thanks for the suggestions. One question tho, I've triggered all the tests locally and it actually passed, what's the difference between the suggested solutions and the one I've now uploaded?

Reply:
Really? Hm, then I'm a bit confused because unless it's a tuple on the right hand side, it should give a boolean array and the error about its truth value being ambiguous

Did you also run the pre-commit? I'm 80 % positive that black will remove the parentheses on the right hand side

Reply:
Yes, ran both pre-commits and the CI tests. The tests were returning the error about truth value, but only when I left the comas:
        assert tuple(pmodel.coords["x"]) == (data.coords["x"],)
         assert tuple(pmodel.coords["y"]) == (data.coords["y"],)
I think I had something else in the parenthesis at the beginning, and forgot to remove the coma. after getting rid of them, both pre-commits and the CI tests were satisfied

Reply:
you're right about the boolean array, it doesn't cause errors tho:
<img width="1126" alt="Zrzut ekranu 2023-02-17 o 15 43 24" src="https://user-images.githubusercontent.com/119355382/219685817-38f21f51-687e-41af-b392-8b84253c274a.png">

Reply:
what if you put an `assert ` at the beginning of the last line?

Does it raise if you use `(-1, -2, -3)` as the left hand side?

Reply:
you're right, when I try to assert it then it complains about ambiguity of comparing boolean arrays. Strange that when running the test suite locally it didn't fail there